app/database/operator/server_data_operator/transformers/channel.ts

```diff
@@ -120,18 +121,20 @@ export const transformChannelInfoRecord = ({action, database, value}: Transforme
  * @param {RecordPair} operator.value
  * @returns {Promise<MyChannelModel>}
  */
-export const transformMyChannelRecord = ({action, database, value}: TransformerArgs): Promise<MyChannelModel> => {
+export const transformMyChannelRecord = async ({action, database, value}: TransformerArgs): Promise<MyChannelModel> => {
     const raw = value.raw as ChannelMembership;
     const record = value.record as MyChannelModel;
     const isCreateAction = action === OperationType.CREATE;
 
+    const isCRTEnabled = await getIsCRTEnabled(database);
+
     const fieldsMapper = (myChannel: MyChannelModel) => {
         myChannel._raw.id = isCreateAction ? (raw.channel_id || myChannel.id) : record.id;
         myChannel.roles = raw.roles;
-        myChannel.messageCount = raw.msg_count;
+        myChannel.messageCount = isCRTEnabled ? raw.msg_count_root! : raw.msg_count;
```

Comment:
I'm almost 99% sure there are some actions in the codebase that modify this values in a db.write and I don't see them modified in this PR